app/src/main/java/ch/epfl/sdp/drone3d/map/MapboxHomeDrawer.kt

```diff
@@ -24,24 +24,28 @@ import com.mapbox.mapboxsdk.utils.ColorUtils
 class MapboxHomeDrawer(mapView: MapView, mapboxMap: MapboxMap, style: Style) : MapboxDrawer {
     private var circleManager = CircleManager(mapView, mapboxMap, style)
     private lateinit var marker: Circle
-    private var reset: Boolean = false
+    private var reset: Boolean = true
 
     /**
      * Draws the [location] of the home on the map
      */
     fun showHome(location: LatLng?) {
-        if (location == null) {
-            circleManager.deleteAll()
-            reset = true
-        } else if (!::marker.isInitialized || reset) {
-            val circleOptions = CircleOptions()
-                .withLatLng(location)
-                .withCircleColor(ColorUtils.colorToRgbaString(Color.RED))
-            marker = circleManager.create(circleOptions)
-            reset = false
-        } else {
-            marker.latLng = location
-            circleManager.update(marker)
+        when {
+            location == null -> {
+                circleManager.deleteAll()
+                reset = true
+            }
+            reset -> {
+                val circleOptions = CircleOptions()
+                    .withLatLng(location)
+                    .withCircleColor(ColorUtils.colorToRgbaString(Color.RED))
+                marker = circleManager.create(circleOptions)
+                reset = false
+            }
+            else -> {
+                marker.latLng = location
+                circleManager.update(marker)
```

Comment:
By curiosity, what do we consider to be the home location during the flight ? Can it be updated ?

Reply:
The home is the lanching point of the drone. It has this name in mavsdk. It can't be changed

Reply:
Ok thanks